di_website/api/utils.py

```diff
@@ -19,9 +21,17 @@ def serialise_page(request, page, fields=['title', 'full_url', 'active']):
     """
     result = object_to_dict(page, fields)
     result['relative_url'] = page.relative_url(request.site, request)
-
     return result
 
+def serialiseDatasources(request, spotlightPage):
+    dataSourceLinks = []
+    for item in spotlightPage.datasources_links:
+        page = {}
+        page['caption'] = item.value['caption']
```

Comment:
The option to use the page title as a caption should be valid here (as I'm hoping u reused `LinkBlock`), in which case you'll have to handle that as well.

Reply:
Now uses the page url if caption is undefined